include/networkit/distance/DynPrunedLandmarkLabeling.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef NETWORKIT_DISTANCE_DYN_PRUNED_LANDMARK_LABELING_HPP_
+#define NETWORKIT_DISTANCE_DYN_PRUNED_LANDMARK_LABELING_HPP_
+
+#include <stdexcept>
+#include <vector>
+
+#include <networkit/base/DynAlgorithm.hpp>
+#include <networkit/distance/PrunedLandmarkLabeling.hpp>
+#include <networkit/dynamics/GraphEvent.hpp>
+#include <networkit/graph/Graph.hpp>
+
+namespace NetworKit {
+
+/**
+ * @ingroup distance
+ * Dynamic pruned landmark labeling.
+ */
+class DynPrunedLandmarkLabeling final : public PrunedLandmarkLabeling, public DynAlgorithm {
+
+public:
+    /**
+     * Dynamic Pruned Landmark Labeling algorithm based on the paper "Fully Dynamic 2-Hop Cover
+     * Labeling " from D'Angelo et al., ACM JEA 2019. The algorithm computes distance labels by
+     * performing pruned breadth-first searches from each vertex. Distance labels can be updated
+     * efficiently after edge insertions.
+     * @note this algorithm only works for unweighted graphs and only supports edge insertions.
```

Comment:
What happens for weighted graphs? This case is not tested currently, so likely no error. From the code I would assume, that weights are just ignored?

Reply:
True, I clarified the comment.